### PR TITLE
Status and Model Control API

### PR DESCRIPTION
**Pull request by tanmayv25**

### CODE REVIEW

File: src/clients/python/experimental_api_v2/examples/simple_client_v2.py [hunk 1]
```diff
@@ -43,26 +43,55 @@
...
+    ### Model Control API ###
+    TRTISClient.load(model_name)
+    if TRTISClient.is_model_ready(model_name, 1):
+        print("PASS: load")
+
+    TRTISClient.load(model_name_2)
+    if TRTISClient.is_model_ready(model_name_2, 1):
+        print("PASS: load_2")
+
+    ### Health API ###
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/examples/simple_client_v2.py]: You should check server live and ready first.

File: src/clients/python/experimental_api_v2/examples/simple_client_v2.py [hunk 2]
```diff
@@ -43,26 +43,55 @@
...
     if TRTISClient.is_server_live():
-        print("SUCCESS: is_server_live")
+        print("PASS: is_server_live")
 
     if TRTISClient.is_server_ready():
-        print("SUCCESS: is_server_ready")
+        print("PASS: is_server_ready")
+
+    if TRTISClient.is_model_ready(model_name, 1):
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/examples/simple_client_v2.py]: This is redundant with the checks in model control part

File: src/clients/python/experimental_api_v2/library/__init__.py [hunk 3]
```diff
@@ -105,5 +209,126 @@ def is_server_ready(self):
             If unable to get readiness.
 
         """
-        self.response = self.client_stub.get("/api/health/ready")
+        self.response = self._client_stub.get("/api/health/ready")
+        self._last_request_id = _raise_if_error(self.response['NV-Status'])
         return self.response.status_code == 200
+
+    def is_model_ready(self, model_name, model_version):
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/library/__init__.py]: model_version should be optional and default to -1, which means that the server will choose the version based on its version policy for the model.  Update docs and simple example appropriately.

File: src/clients/python/experimental_api_v2/library/__init__.py [hunk 4]
```diff
@@ -105,5 +209,126 @@ def is_server_ready(self):
...
+            The version of the model
+
+        Returns
+        -------
+        bool
+            True if the model is ready to be served, False if otherwise.
+
+        Raises
+        ------
+        Exception
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/library/__init__.py]: InferenceServerException

File: src/clients/python/experimental_api_v2/library/__init__.py [hunk 5]
```diff
@@ -105,5 +209,126 @@ def is_server_ready(self):
...
+        """
+
+        self.response = self._client_stub.get("/api/status/" + model_name +
+                                              "?format=json")
+        self._last_request_id = _raise_if_error(self.response['NV-Status'])
+
+        status = json.loads(self.response.read())
+        return status
+
+    def load(self, model_name):
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/library/__init__.py]: load_model

File: src/clients/python/experimental_api_v2/library/__init__.py [hunk 6]
```diff
@@ -105,5 +209,126 @@ def is_server_ready(self):
...
+        """Request the inference server to load specified model.
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/library/__init__.py]: or reload

File: src/clients/python/experimental_api_v2/library/__init__.py [hunk 7]
```diff
@@ -105,5 +209,126 @@ def is_server_ready(self):
...
+        ------
+        Exception
+            If unable to load the model.
+
+        """
+        self.response = self._client_stub.post("/api/modelcontrol/load/" +
+                                               model_name)
+        self._last_request_id = _raise_if_error(self.response['NV-Status'])
+
+    def unload(self, model_name):
```
Comment (deadeyegoodwin) [inline src/clients/python/experimental_api_v2/library/__init__.py]: unload_model

File: src/clients/python/experimental_api_v2/library/__init__.py (hunk 3 above)
Comment (tanmayv25) [reply src/clients/python/experimental_api_v2/library/__init__.py]: The api to retrieve model readiness with or without model version is a part of V2 right? Here as a workaround I am retrieving the model readiness by getting model_status first and then looking for the MODEL_READY value. I have made a note to update when integrating with HTTP V2.

File: src/clients/python/experimental_api_v2/library/__init__.py (hunk 4 above)
Comment (tanmayv25) [reply src/clients/python/experimental_api_v2/library/__init__.py]: Even though I have modified Exception->InferenceServerException, all other kinds of exceptions can be raised by the underlying python modules. Would it be appropriate to document InferenceServerException here? Should we be catching these exceptions and raising them as an InferenceServerException, at the expense of visibility of the stack?

**Pull request merged**